src/LogFileReader.cpp

```diff
@@ -74,12 +81,28 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    QStringList sections = line.split(BatData_Delimiter);
+    if(sections.length() != COLUMNS)
+    {
+       return false;
+    }
+
+    QString timeString = sections.at(0);
+    batteryData.time = QTime::fromString(timeString,STRING_TIME_FORMAT);
+
+    bool voltageOK;
+    batteryData.voltage = sections.at(1).toDouble(&voltageOK);
+
+    bool currentOK;
+    batteryData.current = sections.at(2).toDouble(&currentOK);
+
+    if(voltageOK && currentOK && batteryData.time.isValid())
+    {
+       return true;
+    }
+    else
+    {
+       return false;
+    }
```

Comment:
IIRC, we said error checking would still be on them. Only file parsing without error checking is provided.